GVFS/GVFS.FunctionalTests/Tests/DiskLayoutVersionTests.cs

```diff
@@ -0,0 +1,78 @@
+using GVFS.Tests.Should;
+using NUnit.Framework;
+using System.Runtime.InteropServices;
+
+namespace GVFS.FunctionalTests.Windows.Tests
+{
+    [TestFixture]
+    [Category(Categories.FullSuiteOnly)]
+    public class DiskLayoutVersionTests : TestsWithEnlistmentPerTestCase
+    {
+        private const int WindowsCurrentDiskLayoutMajorVersion = 17;
+        private const int MacCurrentDiskLayoutMajorVersion = 18;
+        private const int WindowsCurrentDiskLayoutMinimumMajorVersion = 7;
+        private const int MacCurrentDiskLayoutMinimumMajorVersion = 18;
+        private const int CurrentDiskLayoutMinorVersion = 0;
+        private int currentDiskMajorVersion;
+        private int currentDiskMinimumMajorVersion;
+
+        [SetUp]
+        public override void CreateEnlistment()
+        {
+            base.CreateEnlistment();
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                this.currentDiskMajorVersion = MacCurrentDiskLayoutMajorVersion;
+                this.currentDiskMinimumMajorVersion = MacCurrentDiskLayoutMinimumMajorVersion;
+            }
+            else
+            {
+                this.currentDiskMajorVersion = WindowsCurrentDiskLayoutMajorVersion;
+                this.currentDiskMinimumMajorVersion = WindowsCurrentDiskLayoutMinimumMajorVersion;
+            }
+        }
+
+        [TestCase]
+        public void MountSucceedsIfMinorVersionHasAdvancedButNotMajorVersion()
+        {
+            // Advance the minor version, mount should still work
+            this.Enlistment.UnmountGVFS();
+            GVFSHelpers.SaveDiskLayoutVersion(
+                this.Enlistment.DotGVFSRoot,
+                this.currentDiskMajorVersion.ToString(),
+                (CurrentDiskLayoutMinorVersion + 1).ToString());
+            this.Enlistment.TryMountGVFS().ShouldBeTrue("Mount should succeed because only the minor version advanced");
+
+            // Advance the major version, mount should fail
+            this.Enlistment.UnmountGVFS();
+            GVFSHelpers.SaveDiskLayoutVersion(
+                this.Enlistment.DotGVFSRoot,
+                (this.currentDiskMajorVersion + 1).ToString(),
+                CurrentDiskLayoutMinorVersion.ToString());
+            this.Enlistment.TryMountGVFS().ShouldBeFalse("Mount should fail because the major version has advanced");
+        }
+
+        [TestCase]
+        public void MountFailsIfBeforeMinimumVersion()
+        {
+            // Advance the minor version, mount should still work
+            this.Enlistment.UnmountGVFS();
+            GVFSHelpers.SaveDiskLayoutVersion(
+                this.Enlistment.DotGVFSRoot,
+                this.currentDiskMinimumMajorVersion.ToString(),
+                (CurrentDiskLayoutMinorVersion).ToString());
+            this.Enlistment.TryMountGVFS().ShouldBeTrue("Mount should succeed because we are using minimum version");
+
+            // Advance the major version, mount should fail

```

Comment:
This comment needs to be updated to something like "Mount should fail if on disk version is below minimum supported version"